File: 735-AsteroidCollision/735-AsteroidCollision.cpp
// Last updated: 6/11/2026, 11:31:43 AM
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& arr) {
        stack<int> st;
        int n = arr.size();
        for(int i=0; i<n; i++){
            int cur = arr[i];
            bool add = 1;
            while(!st.empty() && st.top()>0 && cur<0){
                int rem = cur+st.top();
                if(rem<0){
                    st.pop();
                }else if(rem == 0){
                    add = 0;
                    st.pop();
                    break;
                }else{
                    add = 0;
                    break;
                }
            }
            if(add) st.push(arr[i]);
        }

        vector<int> ans;
        while(!st.empty()){
            ans.push_back(st.top());
            st.pop();
        }

        reverse(ans.begin(), ans.end());
        return ans;


    }
};

File: 735-AsteroidCollision/735-AsteroidCollision.py
# Last updated: 6/11/2026, 11:31:48 AM
class Solution:
    def asteroidCollision(self, arr: List[int]) -> List[int]:
        
        st = []
        i = 0
        while(i<len(arr)):
            print(i)
            if(arr[i]>0):
                st.append(arr[i])
                i=i+1 #mistake-1
            else:
                if(len(st)==0):
                    st.append(arr[i])
                    i+=1
                    continue
                el = st[-1]
                if(el<0):
                    st.append(arr[i])
                    i+=1
                    continue
                elif(abs(el) == abs(arr[i])):
                    st.pop()
                    i+=1
                    continue
                elif(abs(el)>abs(arr[i])):
                    i+=1
                    continue
                else:
                    st.pop()
        print(st)
        return st

                